ext/semian/semian.c

```diff
@@ -221,79 +161,7 @@ create_semaphore(int key, int permissions, int *created)
   return semid;
 }
 
-static int
-get_semaphore(int key)
-{
-  return semget(key, SI_NUM_SEMAPHORES, 0);
-}
-
-/*
- * call-seq:
- *    Semian::Resource.new(id, tickets, permissions, default_timeout) -> resource
- *
- * Creates a new Resource. Do not create resources directly. Use Semian.register.
- */
-static VALUE
-semian_resource_initialize(VALUE self, VALUE id, VALUE tickets, VALUE permissions, VALUE default_timeout)
-{
-  key_t key;
-  int created = 0;
-  semian_resource_t *res = NULL;
-  const char *id_str = NULL;
-
-  if (TYPE(id) != T_SYMBOL && TYPE(id) != T_STRING) {
```

Comment:
Note that I refactored these type checks into static functions within resource.c.

They now also cast to the correct type. In the existing code there is actually causing truncation in some places where a long is being downcast to an int.